### PR TITLE
Update dependencies [jan 29]

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by ebyhr** [comment]
Could you fix another error prone failure? 
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project trino-main: Compilation failure
Error:  /home/runner/work/trino/trino/core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java:[3074,33] [UnusedVariable] The parameter 'event' is never read.
```

**Comment by wendigo** [comment]
@ebyhr PTAL

**Comment by wendigo** [comment]
@ebyhr ptal

File: core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java [hunk 1]
```diff
@@ -3071,6 +3071,7 @@ default T onSinkInstanceHandleAcquired(SinkInstanceHandleAcquiredEvent event)
             return onEvent(event);
         }
 
+        @SuppressWarnings("UnusedVariable")
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java:3074]: Don't suppress. I think you can just name the variable "ignored" to please the checker.

File: core/trino-spi/src/test/java/io/trino/spi/testing/TestInterfaceTestUtils.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+@SuppressWarnings("UnusedVariable")
```
Comment (martint) [inline core/trino-spi/src/test/java/io/trino/spi/testing/TestInterfaceTestUtils.java:22]: Don't suppress warnings. It hides potential legitimate problems and should only be used as a last resort.

File: lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java [hunk 3]
```diff
@@ -278,7 +278,7 @@ private static void assertLocation(
             assertThat(uriIncompatibilities).isEqualTo(Set.of("skipped"));
         }
 
-        assertThat(location).isEqualTo(location);
+        assertThat(location.equals(location)).isTrue();
```
Comment (martint) [inline lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java:281]: What's wrong with the previous form?

File: lib/trino-orc/src/test/java/io/trino/orc/metadata/statistics/AbstractRangeStatisticsTest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ private void assertMinMaxStatistics(T min, T max)
         assertThat(statistics.getMin()).isEqualTo(min);
         assertThat(statistics.getMax()).isEqualTo(max);
 
-        assertThat(statistics).isEqualTo(statistics);
+        assertThat(statistics.equals(statistics)).isTrue();
```
Comment (martint) [inline lib/trino-orc/src/test/java/io/trino/orc/metadata/statistics/AbstractRangeStatisticsTest.java:55]: Same here.

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/TestLoggingInvocationHandler.java [hunk 5]
```diff
@@ -95,6 +95,7 @@ public String run(boolean ok, String s)
 
     private interface SomeInterface
     {
+        @SuppressWarnings("UnusedVariable")
```
Comment (martint) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/TestLoggingInvocationHandler.java:98]: Don't suppress warnings. Rename the variables, instead.

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/util/TestPrioritizedFifoExecutor.java [hunk 6]
```diff
@@ -149,4 +146,10 @@ private void testBound(int maxThreads, int totalTasks)
 
         assertThat(failed.get()).isFalse();
     }
+
+    @SuppressWarnings("UnusedVariable")
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/util/TestPrioritizedFifoExecutor.java:150]: Same here.

File: lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java (hunk 3 above)
Comment (wendigo) [reply lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java:281]: It violates an error-prone check: https://errorprone.info/bugpattern/TruthSelfEquals

File: lib/trino-orc/src/test/java/io/trino/orc/metadata/statistics/AbstractRangeStatisticsTest.java (hunk 4 above)
Comment (wendigo) [reply lib/trino-orc/src/test/java/io/trino/orc/metadata/statistics/AbstractRangeStatisticsTest.java:55]: https://errorprone.info/bugpattern/TruthSelfEquals

**Comment by wendigo** [comment]
@martint PTAL

File: lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java (hunk 3 above)
Comment (wendigo) [reply lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java:281]: Yes. I figured that this is about to test equals implementation. Maybe in these cases we should use a dedicated `equals` and `hashCode` testers? (i.e. https://github.com/jqno/equalsverifier)

Comment (martint) [reply lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java:281]: We have something like that in Airlift: EquivalenceTester

**Comment by wendigo** [comment]
@martint I've used `equivalenceTester` to fix `TruthSelfEquals` error-prone violations

**Review by losipiuk — APPROVED** [review]

**Pull request merged**